awx/main/utils/pglock.py

```diff
@@ -8,9 +8,22 @@
 
 
 @contextmanager
-def advisory_lock(*args, **kwargs):
+def advisory_lock(*args, lock_session_timeout_milliseconds=0, **kwargs):
     if connection.vendor == 'postgresql':
+        cur = None
+        idle_in_transaction_session_timeout = None
+        idle_session_timeout = None
+        if lock_session_timeout_milliseconds > 0:
+            cur = connection.cursor()
+            idle_in_transaction_session_timeout = cur.execute('SHOW idle_in_transaction_session_timeout').fetchone()[0]
+            idle_session_timeout = cur.execute('SHOW idle_session_timeout').fetchone()[0]
+            cur.execute(f"SET idle_in_transaction_session_timeout = {lock_session_timeout_milliseconds}")
+            cur.execute(f"SET idle_session_timeout = {lock_session_timeout_milliseconds}")
         with django_pglocks_advisory_lock(*args, **kwargs) as internal_lock:
             yield internal_lock
+            if lock_session_timeout_milliseconds > 0:
+                cur.execute(f"SET idle_in_transaction_session_timeout = {idle_in_transaction_session_timeout}")
```

Comment:
in the case that we are closed, this will hit a `InterfaceError: the cursor is closed` error

in the case of task managers, we probably don't care because we are willing for the entire pid to go away, but if someone later on uses `lock_session_timeout_milliseconds` in another case there might be adverse affects

Reply:
cursor has a `closed` property that we can condition off of

Reply:
yah these 2 parts shouldnt have to shared a cursor... let me go ahead and use context manager for these 2 section of code